openhands/integrations/bitbucket/bitbucket_service.py

```diff
@@ -0,0 +1,371 @@
+                # If both approaches fail, log the error and return an empty list
+                from openhands.core.logger import openhands_logger as logger
+
+                logger.warning(
+                    f'Failed to get repositories from Bitbucket: {e}, then: {nested_e}'
+                )
+
+        return repositories
+
+    async def get_suggested_tasks(self) -> list[SuggestedTask]:
+        """Get suggested tasks for the authenticated user across all repositories."""
+        # This would require multiple API calls to get PRs with conflicts, failing checks, etc.
+        # For now, we'll return an empty list
+        return []
+
+    async def get_repository_details_from_repo_name(
+        self, repository: str
+    ) -> Repository:
+        """Gets all repository details from repository name."""
+        # Extract owner and repo from the repository string (e.g., "owner/repo")
+        parts = repository.split('/')
+        if len(parts) < 2:
+            raise ValueError(f'Invalid repository name: {repository}')
+
+        owner = parts[-2]
+        repo = parts[-1]
+
+        url = f'{self.BASE_URL}/repositories/{owner}/{repo}'
+        data, _ = await self._make_request(url)
+
+        # Convert UUID to an integer hash for compatibility with Repository model
+        uuid = data.get('uuid', '')
+        # Remove curly braces if present and use hash of the UUID string as the ID
+        if uuid.startswith('{') and uuid.endswith('}'):
+            uuid = uuid[1:-1]
+        id_value = hash(uuid) % (2**31)  # Ensure it's a positive 32-bit integer
+
+        return Repository(
+            id=id_value,
+            full_name=f'{data.get("workspace", {}).get("slug", "")}/{data.get("slug", "")}',
+            git_provider=ProviderType.BITBUCKET,
+            is_public=data.get('is_private', True) is False,
+            stargazers_count=None,  # Bitbucket doesn't have stars
+            pushed_at=data.get('updated_on'),
+        )
+
+    async def get_branches(self, repository: str) -> list[Branch]:
+        """Get branches for a repository."""
+        # Extract owner and repo from the repository string (e.g., "owner/repo")
+        parts = repository.split('/')
+        if len(parts) < 2:
+            raise ValueError(f'Invalid repository name: {repository}')
+
+        owner = parts[-2]
+        repo = parts[-1]
+
+        url = f'{self.BASE_URL}/repositories/{owner}/{repo}/refs/branches'
+        data, _ = await self._make_request(url)
+
+        branches = []
+        for branch in data.get('values', []):
+            branches.append(
+                Branch(
+                    name=branch.get('name', ''),
+                    commit_sha=branch.get('target', {}).get('hash', ''),
+                    protected=False,  # Bitbucket doesn't expose this in the API
+                    last_push_date=branch.get('target', {}).get('date', None),
+                )
+            )
+
+        return branches
+
+    async def create_pr(
```

Comment:
If bitbucket supports opening draft PRs, we should expose that param to the agent via the MCP tool and plumbed it through here